utils/hct/gen_intrin_main.txt

```diff
@@ -339,8 +339,8 @@ float<4,3> [[rn]] ObjectToWorld4x3();
 float<4,3> [[rn]] WorldToObject4x3();
 
 // Packed dot products with accumulate:
-$type3 [[rn]] dot4add_u8packed(in uint a, in $type1 b, in uint c);
-$type3 [[rn]] dot4add_i8packed(in uint a, in $type1 b, in int c);
+uint [[rn]] dot4add_u8packed(in uint a, in $type1 b, in uint c);
+int [[rn]] dot4add_i8packed(in uint a, in $type1 b, in int c);
 $type3 [[rn]] dot2add(in float16_t<2> a, in $type1 b, in float c);
```

Comment:
Give that `dot2add`  has the same issue, it should also be fixed and tested as well. You'll need to `-enable-16bit-types` in the test though to avoid an unrelated assert that you could very well fix, but should probably be part of a separate PR.

Reply:
Thanks for the warning about `-enable-16bit-types`.